### PR TITLE
Add the concoct cut_up_fasta tool

### PR DESCRIPTION
**Pull request by gregvonkuster**
FOR CONTRIBUTOR:
* [x] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [x] - License permits unrestricted use (educational + commercial)
* [x] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/concoct/cut_up_fasta.xml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        </param>
+    </inputs>
+    <outputs>
+        <data name="output_bed" format="bed" label="${tool.name} on ${on_string} (bed)">
+            <filter>output_bed_param == 'yes'</filter>
+        </data>
+        <data name="output_fasta" format="fasta" label="${tool.name} on ${on_string} (contigs)"/>
+    </outputs>
+    <tests>
+        <test expect_num_outputs="2">
```
Comment (bgruening) [inline tools/concoct/cut_up_fasta.xml:38]: @gregvonkuster should this tool also have test for the other options and different combinations - kindof to also test the Python code.

File: tools/concoct/cut_up_fasta.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        _open = open
+
+    if bedoutfile is not None:
+        bedoutfile_fh = open(bedoutfile, 'w')
+
+    with _open(input_fasta) as fh:
+        for record in SeqIO.parse(fh, "fasta"):
+            if (not merge_last and len(record.seq) > chunk_size) or (merge_last and len(record.seq) >= 2 * chunk_size):
+                i = 0
+                for split_seq in chunks(record.seq, chunk_size, overlap, merge_last):
```
Comment (bgruening) [inline tools/concoct/cut_up_fasta.py:23]: enumerate?

File: tools/concoct/cut_up_fasta.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+                    print(">%s.%i\n%s" % (record.id, i, split_seq))
+                    if bedoutfile is not None:
+                        print("{0}\t{2}\t{3}\t{0}.{1}".format(record.id, i, chunk_size * i, chunk_size * i + len(split_seq)), file=bedoutfile_fh)
+                    i = i + 1
+            else:
+                print(">%s\n%s" % (record.id, record.seq))
```
Comment (bgruening) [inline tools/concoct/cut_up_fasta.py:29]: f-strings?

File: tools/concoct/cut_up_fasta.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        bedoutfile_fh = open(bedoutfile, 'w')
+
+    with _open(input_fasta) as fh:
+        for record in SeqIO.parse(fh, "fasta"):
+            if (not merge_last and len(record.seq) > chunk_size) or (merge_last and len(record.seq) >= 2 * chunk_size):
+                i = 0
+                for split_seq in chunks(record.seq, chunk_size, overlap, merge_last):
+                    print(">%s.%i\n%s" % (record.id, i, split_seq))
+                    if bedoutfile is not None:
+                        print("{0}\t{2}\t{3}\t{0}.{1}".format(record.id, i, chunk_size * i, chunk_size * i + len(split_seq)), file=bedoutfile_fh)
```
Comment (bgruening) [inline tools/concoct/cut_up_fasta.py:26]: I think bedoutfile_fh.write() is the preferred way

File: tools/concoct/cut_up_fasta.xml (hunk 1 above)
Comment (gregvonkuster) [reply tools/concoct/cut_up_fasta.xml:50]: I've added more tests.

File: tools/concoct/cut_up_fasta.py (hunk 4 above)
Comment (gregvonkuster) [reply tools/concoct/cut_up_fasta.py]: Fixed -thanks!

File: tools/concoct/cut_up_fasta.py (hunk 3 above)
Comment (gregvonkuster) [reply tools/concoct/cut_up_fasta.py]: Fixed - thanks!

**Comment by gregvonkuster** [comment]
I'm still fixing some issues on this - I should have another commit soon.

**Comment by gregvonkuster** [comment]
This should now be ready if tests pass.

**Comment by gregvonkuster** [comment]
This is ready for review when convenient - there's now 3 tools instead of 1 though... :)

File: tools/concoct/coverage_table.xml [hunk 5]
```diff
@@ -0,0 +1,40 @@
+<tool id="concoct_coverage_table" name="CONCOCT coverage table" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="@PROFILE@">
+    <description></description>
```
Comment (bgruening) [inline tools/concoct/coverage_table.xml:2]: description missing?

Comment (gregvonkuster) [reply tools/concoct/coverage_table.xml:2]: I think CONCOCT coverage table relays what is needed, so not sure what additional description is needed.  I'm happy to add text to the description if you have an idea though.

**Review by bgruening — APPROVED** [review]
Ok, thanks @gregvonkuster

**Pull request merged**